### ISSUE TITLE
valid-types cannot parse links right after another tag

### ISSUE THREAD

**Issue by LukeAbby** [gajus/eslint-plugin-jsdoc#1381]
## Expected behavior
I expected `@remarks {@link SomeType}` to parse.

## Actual behavior
I get an error saying it's invalid. I considered posting this issue on https://github.com/jsdoc-type-pratt-parser based upon the guidance in this issue format but I don't believe it's part of the concern of that library. It's probably being given an erroneous input.

## ESLint Config

```json
{
    "jsdoc/valid-types": "error"
}
```

## ESLint sample

```ts
type SomeType = number;

// Ok
/** {@link SomeType} */

// Ok
/** @remarks foo bar lorem ipsum {@link SomeType} */

// Not ok
/** @remarks {@link SomeType} */
```

## Environment

- Node version: v20.15.0
- ESLint version v9.26.0
- `eslint-plugin-jsdoc` version: ^50.6.11
